instrumentation/spring/spring-boot-resources/library/build.gradle.kts

```
@@ -0,0 +1,12 @@
+plugins {
+  id("otel.library-instrumentation")
+}
+
+dependencies {
+  compileOnly("io.opentelemetry:opentelemetry-sdk-extension-autoconfigure-spi")
+  annotationProcessor("com.google.auto.service:auto-service")
+  compileOnly("com.google.auto.service:auto-service-annotations")
+  implementation("org.yaml:snakeyaml:1.31")
+  testCompileOnly("com.google.auto.service:auto-service-annotations")
+  testImplementation("io.opentelemetry:opentelemetry-sdk-extension-autoconfigure-spi")
```

Nit: I'd split these dependencies into several groups, to make them a bit more readable
```suggestion
  compileOnly("io.opentelemetry:opentelemetry-sdk-extension-autoconfigure-spi")
  
  annotationProcessor("com.google.auto.service:auto-service")
  compileOnly("com.google.auto.service:auto-service-annotations")
  testCompileOnly("com.google.auto.service:auto-service-annotations")
  
  implementation("org.yaml:snakeyaml:1.31")
  
  testImplementation("io.opentelemetry:opentelemetry-sdk-extension-autoconfigure-spi")
```